src/ott/datasets.py

```diff
@@ -105,6 +110,168 @@ def _create_sample_generators(self) -> Iterator[jnp.array]:
+      self.rotation_fn = lambda x: jnp.matmul(rotation_marix, x)
+
+  def _create_sample_generators(self) -> Iterator[jnp.ndarray]:
+    rng = jax.random.PRNGKey(0) if self.init_rng is None else self.init_rng
+
+    while True:
+      rng, _ = jax.random.split(rng)
+      seed = jax.random.randint(rng, [], minval=0, maxval=1e5).item()
+      if self.name == "moon":
+        samples, _ = sklearn.datasets.make_moons(
+            n_samples=(self.batch_size, 0),
+            random_state=seed,
+            noise=self.std_noise,
+        )
+      elif self.name == "s_curve":
+        x, _ = sklearn.datasets.make_s_curve(
+            n_samples=self.batch_size,
+            random_state=seed,
+            noise=self.std_noise,
+        )
+        samples = x[:, [2, 0]] if self.dim_data == 2 else x
+      elif self.name == "swiss":
+        x, _ = sklearn.datasets.make_swiss_roll(
+            n_samples=self.batch_size,
+            random_state=seed,
+            noise=self.std_noise,
+        )
+        samples = x[:, [2, 0]] if self.dim_data == 2 else x
+      else:
+        raise NotImplementedError(
+            f"SklearnDistribution `{self.name}` not implemented."
+        )
+
+      samples = jnp.asarray(samples, dtype=jnp.float32)
+      samples = jax.vmap(self.rotation_fn)(samples)
+      samples = self._offset + self.scale * samples
+      yield samples
+
+
+@dataclasses.dataclass
+class SortedSprial:
+  """A class to define 2 or 3-dimensional ordered spiral distributions.
+
+  The spiral is sorted in the sense that the indices of the points in each
+  generated batch follow the progression of the spiral, i.e. the angles to draw
+  the points are linearly increasing between ``min_angle`` and ``max_angle``.
+  Afterwards, the first point of a batch point always has a norm close to
+  ``min_radius``, while the last point of the batch always has a norm close
+  to ``max_radius``, and the intermediate points have a linearly increasing
+  norm between ``min_radius`` and ``max_raidus``.
+
+  Args:
+    init_rng: initial PRNG key
+    dim_data: data dimensionality
+    min_angle: angle we start from to draw the spiral
+    max_angle: angle of the spiral's ending point
+    std_noise: standard deviation of the gaussian additive noise
+    batch_size: batch size of the samples
+  """
+
+  init_rng: jax.Array
+  dim_data: int = 2
+  min_radius: float = 3.
+  max_radius: float = 10.
+  min_angle: float = 0.
+  max_angle: float = 10.
+  std_noise: float = 0.01
+  batch_size: int = 1024
+
+  def __iter__(self) -> Iterator[jnp.ndarray]:
+    """Random sample generator from an ordered spiral distribution.
+
+    Returns:
+    A generator of samples from the ordered spiral distribution.
+    """
+    return self._create_sample_generators()
+
+  def _create_sample_generators(self) -> Iterator[jnp.ndarray]:
+    rng = jax.random.PRNGKey(0) if self.init_rng is None else self.init_rng
+
+    while True:
+      rng, rng1, rng2 = jax.random.split(rng, 3)
```

Comment:
Maybe gives the `rng1` and `rng2` more meaningful names, like `rng_noise` and `rng_3rd_axis`.